src/telliot_feeds/sources/mimicry.py

```diff
@@ -0,0 +1,259 @@
+from dataclasses import dataclass
+import operator
+from typing import Any, List, Union
+from typing import Optional
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+from telliot_feeds.datasource import DataSource
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.utils.log import get_logger
+
+
+
+logger = get_logger(__name__)
+retry_strategy = Retry(
+    total=3,
+    backoff_factor=1,
+    status_forcelist=[429, 500, 502, 503, 504],
+    allowed_methods=["GET"],
+)
+adapter = HTTPAdapter(max_retries=retry_strategy)
+
+@dataclass
+class Transaction:
+    """wrapper class for nft sale transactions"""
+    price: int
+    item_id: Union[int, str]
```

Comment:
Why can the item ids be both `str` & `int`? Why not enforce one?

Reply:
Ah, this is because some item ids for NFTS will be strings (e.g. "Hyacinth"), but in other cases they will be integers. This is also the type enforcement from the typescript implementation